addons/account/views/report_payment_receipt_templates.xml

```diff
@@ -33,47 +33,80 @@
                          </div>
                     </div>
                 </div>
-                <table class="table table-sm">
-                    <thead>
-                        <tr>
-                            <th><span>Invoice Date</span></th>
-                            <th><span>Invoice Number</span></th>
-                            <th><span>Reference</span></th>
-                            <th class="text-right"><span>Amount</span></th>
-                        </tr>
-                    </thead>
-                    <tbody>
-                        <t t-set="partials" t-value="o.move_id._get_reconciled_invoices_partials()"/>
-                        <t t-set="invoices" t-value="{partial[2].move_id for partial in partials}"/>
-                        <t t-foreach="invoices" t-as="inv">
+
+                <t t-set="partials" t-value="o.move_id._get_reconciled_invoices_partials()"/>
+                <t t-set="invoices" t-value="{partial[2].move_id for partial in partials}"/>
+                <t t-foreach="invoices" t-as="inv">
+                    
+                    <t t-set="currencyInvoice" t-value="inv.currency_id"/>
+
+                    <table class="table table-sm">
+                        <thead>
+                            <tr>
+                                <th><span>Invoice Date</span></th>
+                                <th><span>Invoice Number</span></th>
+                                <th><span>Reference</span></th>
+                                <!-- Check if the invoice includes different currencies -->
+                                <t t-set="otherCurrency" t-value="False"/>
+                                <t t-foreach="inv._get_reconciled_invoices_partials()" t-as="par">
+                                    <t t-set="currencyPayment" t-value="par[2].currency_id"/>
+                                    <t t-if="not currencyPayment == currencyInvoice">
+                                        <t t-set="otherCurrency" t-value="True"/>
+                                    </t>
+                                </t>
```

Comment:
`<t t-set="otherCurrency" t-value="any(currencyPayment != par[2].currency_id for par in inv._get_reconciled_invoices_partials())"/>`